doc/source/whatsnew/v1.6.0.rst

```diff
@@ -158,7 +158,7 @@ Bug fixes
 
 Categorical
 ^^^^^^^^^^^
--
+- Bug in :meth:`Categorical.set_categories` losing dtype information (:issue:`48812`)
```

Comment:
Might be good to mention specifically for pandas / Extension types here

Reply:
It applies to every dtype that is not the default dtype of the constructor, e.g.

```
cat = pd.Categorical(Series([1, 2], dtype="int16"))
ser = Series([4], dtype="int16")
result = cat.add_categories(ser)
```

this falls back to ``int64``. Since this is exactly the same case as above, I did not add tests for this though